testsuite/tests/singlecluster/authorino/authorization/spicedb/test_spicedb.py

```diff
@@ -3,96 +3,124 @@
+cover authentication and are only focused on authorization.
 """
 
 import pytest
-from dynaconf import ValidationError
 
-from testsuite.spicedb.spicedb import SpiceDBService, SchemaConfig, RelationshipConfig
+from testsuite.gateway.gateway_api.route import HTTPRoute
 from testsuite.kubernetes.secret import Secret
+from testsuite.spicedb.spicedb import SpiceDB, SchemaConfig, RelationshipConfig
 
 pytestmark = [pytest.mark.authorino]
 
 
 @pytest.fixture(scope="module")
 def schema_config():
     """Create schema config with custom data."""
-    schema_config = SchemaConfig(
+    return SchemaConfig(
         subject_type="user",
         resource_type="document",
         permission1="read",
         permission2="write",
         relation1="reader",
         relation2="writer",
     )
-    return schema_config
 
 
 @pytest.fixture(scope="module")
 def relationship_config():
     """Create relationship config with custom data."""
-    relationship_config = RelationshipConfig(
+    return RelationshipConfig(
         subject_type="user",
         resource_type="document",
         relations=["writer", "reader"],
         resource_id="document1",
         subject_ids=["admin", "some_user"],
     )
-    return relationship_config
 
 
 @pytest.fixture(scope="module")
-def spicedb_info(testconfig, skip_or_fail):
+def spicedb(request, cluster, blame, module_label, testconfig):
     """
-    Validates that only the 'spicedb' section of the test config is used,
-    and returns the relevant configuration (e.g., URL, secret, etc.).
+    Deploy SpiceDB instance and manage its lifecycle.
+    After the tests, it clears all relationships.
+    """
+    spicedb = SpiceDB(
+        cluster,
+        blame("spicedb"),
+        module_label,
+        image=testconfig["spicedb"]["image"],
+    )
+    request.addfinalizer(spicedb.delete)
+    spicedb.commit()
+    spicedb.wait_for_ready()
+
+    yield spicedb
+
+
+@pytest.fixture(scope="module")
+def spicedb_hostname(exposer, gateway, blame):
+    """Hostname for SpiceDB"""
+    return exposer.expose_hostname(blame("spicedb-hostname"), gateway)
+
+
+@pytest.fixture(scope="module")
+def spicedb_route(request, gateway, spicedb, blame, spicedb_hostname, module_label, cluster):
     """
-    try:
-        testconfig.validators.validate(only="spicedb")
-    except (KeyError, ValidationError) as exc:
-        skip_or_fail(f"Spicedb configuration item is missing: {exc}")
-    return testconfig["spicedb"]
+    Create an HTTPRoute to SpiceDB through the Gateway for external access from test suite.
+    """
+    route = HTTPRoute.create_instance(cluster, blame("spicedb"), gateway, {"app": module_label})
+    route.add_hostname(spicedb_hostname.hostname)
+    route.add_backend(spicedb)
+    request.addfinalizer(route.delete)
+    route.commit()
+
+    spicedb.set_gateway_url(f"http://{spicedb_hostname.hostname}")
+
+    return route
```

Comment:
You dont need to route spicedb client through the gateway and httproute. Openshift Route is enough